vips/vip-128/vip-128.ts

```diff
@@ -0,0 +1,176 @@
+      Reserve Factor: 10%
+      Borrow Cap: 600,000 TUSD
+      Supply Cap: 1,000,000 TUSD
+      OLD VTUSD market:
+      Collateral Factor: 55%
+      Reserve factor: 100%
+      New symbol: vTUSDOLD
+      New name: Venus TUSDOLD
+    `,
+    forDescription: "I agree that Venus Protocol should proceed with the TUSD contract migration",
+    againstDescription: "I do not think that Venus Protocol should proceed with the TUSD contract migration",
+    abstainDescription: "I am indifferent to whether Venus Protocol proceeds with the TUSD contract migration or not",
+  };
+
+  return makeProposal(
+    [
+      {
+        target: COMPTROLLER,
+        signature: "_setActionsPaused(address[],uint8[],bool)",
+        params: [[OLD_VTUSD], [Actions.MINT, Actions.BORROW, Actions.ENTER_MARKETS], true],
+      },
+      {
+        target: OLD_VTUSD,
+        signature: "_setImplementation(address,bool,bytes)",
+        params: [VTUSD_RESETTER, false, "0x"],
+      },
+      {
+        target: OLD_VTUSD,
+        signature: "_setImplementation(address,bool,bytes)",
+        params: [VTOKEN_IMPLEMENTATION, false, "0x"],
+      },
+      {
+        target: OLD_VTUSD,
+        signature: "_setReserveFactor(uint256)",
+        params: [parseUnits("1", 18)],
+      },
+
+      {
+        target: COMPTROLLER,
+        signature: "_setCollateralFactor(address,uint256)",
+        params: [OLD_VTUSD, parseUnits("0.55", 18)],
+      },
+
+      {
+        target: COMPTROLLER,
+        signature: "_supportMarket(address)",
+        params: [NEW_VTUSD],
+      },
+
+      {
+        target: COMPTROLLER,
+        signature: "_setMarketSupplyCaps(address[],uint256[])",
+        params: [[NEW_VTUSD], [parseUnits("1000000", 18)]],
+      },
+
+      {
+        target: COMPTROLLER,
+        signature: "_setMarketBorrowCaps(address[],uint256[])",
+        params: [[NEW_VTUSD], [parseUnits("600000", 18)]],
+      },
+
+      {
+        target: COMPTROLLER,
+        signature: "_setVenusSpeeds(address[],uint256[],uint256[])",
+        params: [[NEW_VTUSD], ["217013888888889"], ["217013888888889"]],
+      },
+
+      {
+        target: CHAINLINK_ORACLE,
+        signature: "setTokenConfig((address,address,uint256))",
+        params: [[NEW_TUSD, ORACLE_FEED, MAX_STALE_PERIOD]],
+      },
+
+      {
+        target: RESILIENT_ORACLE,
+        signature: "setTokenConfig((address,address[3],bool[3]))",
+        params: [
+          [
+            NEW_TUSD,
+            [
+              CHAINLINK_ORACLE,
+              "0x0000000000000000000000000000000000000000",
+              "0x0000000000000000000000000000000000000000",
+            ],
+            [true, false, false],
+          ],
+        ],
+      },
+
+      {
+        target: TREASURY,
+        signature: "withdrawTreasuryBEP20(address,uint256,address)",
+        params: [NEW_TUSD, INITIAL_FUNDING, NORMAL_TIMELOCK],
+      },
+
+      {
+        target: NEW_TUSD,
+        signature: "approve(address,uint256)",
```

Comment:
Do we need to approve first 0?

Reply:
Fixed [here](https://github.com/VenusProtocol/vips/pull/25/commits/53a8b05086695161dc1fa2057382b4db9f292e48)